index.js

```diff
@@ -1176,10 +1176,11 @@
     }
   }
 
-  //# equals :: (a, b) -> Boolean
+  //# equals :: Setoid a => (a, a) -> Boolean
   //.
   //. Returns `true` if its arguments are of the same type and equal according
-  //. to the type's [`fantasy-land/equals`][] method; `false` otherwise.
+  //. to the type's [`fantasy-land/equals`][] method. Returns `false` for
+  //. inputs of differing types.
```

Comment:
I don't think this change is necessary, as we already include the clause _if its arguments are of the same type_.

Reply:
But `false otherwise` is too broad: When a user provides two values of the same type that don't have Setoid, the function throws.

Reply:
We do not typically document the behaviour of a function when applied to an argument outside its domain. The description of [`S.trim`][1], for example, does not consider the possibility of application to a non-string value.


[1]: https://sanctuary.js.org/#trim

Reply:
So you'd prefer to remove the *`false` otherwise* part altogether, and treat the special treatment of inputs of differing types as an undocumented "feature", or simply garbage?

Reply:
> when applied to an argument outside its domain

Supplying two values of differing types are within the domain of the function. So are supplying two Setoids of the same type. Anything else isn't.

Reply:
So I've documented purely the behaviour of the function for values *within* its domain. I have removed documentation that overlaps with values outside of the domain.

Reply:
I do not see how `Setoid a => (a, a) -> Boolean` permits _inputs of differing types_. :thinking:

Reply:
But the *real* signature is:

```hs
equals :: Setoid a => (a, a) -> Boolean
equals :: (a, AnythingBut a) -> Boolean
```

Remember..? But you said that is not pedagogical.

Reply:
The function has a special code path to facilitate the latter overload.

Reply:
That makes these inputs part of the domain in my book.

Reply:
You yourself said:

> We can document in other ways how the function behaves when given values of different types. -- https://github.com/sanctuary-js/sanctuary-type-classes/pull/147#issuecomment-824666181

That's what I'm trying to do here.